frontend/views/containers/chatroom/MessageActions.vue

```diff
@@ -0,0 +1,160 @@
+<template lang='pug'>
+menu-parent(ref='menu')
+  .c-actions
+    button.hide-touch.is-icon-small(
+      :aria-label='L("Add reaction")'
+      @click='action("openEmoticon", $event)'
+    )
+      i.icon-smile-beam
+
+    button.hide-touch.is-icon-small(
+      v-if='isCurrentUser'
+      :aria-label='L("Edit")'
+      @click='action("edit")'
+    )
+      i.icon-pencil-alt
+
+    button.hide-touch.is-icon-small(
+      :aria-label='L("Reply")'
+      @click='action("reply")'
+    )
+      i.icon-reply
+
+    button.hide-touch.is-icon-small(
+      v-if='variant === "failed"'
+      :aria-label='L("Add emoticons")'
+      @click='action("retry")'
+    )
+      i.icon-undo
+
+    menu-trigger.is-icon-small(
+      :aria-label='L("More options")'
+    )
+      i.icon-ellipsis-h
+
+  menu-content.c-content
+    ul
+      menu-item.hide-desktop.is-icon-small(tag='button'
+        @click='action("openEmoticon", $event)'
+      )
+        i.icon-smile-beam
+        | {{L('Add reaction')}}
+
+      menu-item.hide-desktop.is-icon-small(tag='button'
+        v-if='isCurrentUser'
+        @click='action("edit")'
+      )
+        i.icon-pencil-alt
+        | {{L('Edit')}}
```

Comment:
Do all of these need to be written this way, or can they be written on a single line using `i18n`? E.g.:

```pug
i18n.icon-pencil-alt(tag='i') Edit
```